source/common/router/resp_header_parser.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <list>
+#include <string>
+
+#include "envoy/http/header_map.h"
+
+#include "common/protobuf/protobuf.h"
+
+#include "api/base.pb.h"
+
+namespace Envoy {
+namespace Router {
+
+class ResponseHeaderParser;
+typedef std::unique_ptr<ResponseHeaderParser> ResponseHeaderParserPtr;
+
+/**
+ * This class holds the logic required to apply response headers additions and removals.
+ */
+class ResponseHeaderParser {
```

Comment:
Maybe not worth it but is it worthwhile to have a header parser base class? I kind if suspect that as we inevitably add header substitution in the future there will be more and more code overlap.

Reply:
I agree, but I haven't really thought about what that looks like. May if I drop a todo in here to combine them when header substitution gets added?

Reply:
Sure that's fine. Just add a TODO.

Reply:
Done, but I put it in resp_header_parser.cc.